corehq/apps/style/static/style/js/bootstrap3/ui-element.js

```diff
@@ -0,0 +1,479 @@
+                    }
+                    return this;
+                },
+            };
+            return function (guid, modal_title) {
+                return new KeyValList(guid, modal_title);
+            };
+        }()),
+        input_map: (function() {
+            var InputMap = function (show_del_button) {
+                var that = this;
+                eventize(this);
+                this.ui = $('<div class="form-group hq-input-map" />');
+                this.value = {
+                    key: "",
+                    val: "",
+                };
+                this.edit = true;
+                this.show_delete = show_del_button;
+                this.on('change', function() {
+                    this.val(this.ui.find(".enum-key").val(), this.ui.find(".enum-value").val())
+                });
+                this.on('remove', function() {
+                    this.ui.remove();
+                });
+
+                this.$edit_view = $('<div class="form-inline" style="margin-left:5px;" />');
+                var key_input = $('<input type="text" class="form-control enum-key" style="width:220px;" placeholder="key" />'),
+                    val_input = $('<input type="text" class="form-control enum-value" style="width:220px;" placeholder="value" />');
+                key_input.change(function () {
+                    that.fire('change');
+                });
+                val_input.change(function() {
+                    that.fire('change');
+                });
+                this.$edit_view.append(key_input);
+                this.$edit_view.append(' <i class="fa fa-arrow-right"></i> ');
+                this.$edit_view.append(val_input);
+                if(this.show_delete) {
+                    var $deleteButton = $('<a href="#" data-enum-action="remove" class="btn btn-danger" />');
+                    $deleteButton.append('<i class="fa fa-remove"></i> Delete');
+                    $deleteButton.click(function() {
+                        that.fire('remove');
+                        return false;
+                    });
+                    this.$edit_view.append(' ');
+                    this.$edit_view.append($deleteButton);
+                }
+                this.$noedit_view = $('<div />');
+
+                this.setEdit(this.edit);
+            };
+            InputMap.prototype = {
+                val: function(map_key, map_val, translated_map_val) {
+                    if (map_key === undefined) {
+                        return this.value;
+                    } else {
+                        this.value = {
+                            key: map_key,
+                            val: map_val,
+                        };
+                        this.$edit_view.find(".enum-key").val(map_key);
+                        this.$edit_view.find(".enum-value").val(map_val);
+                        if (map_val == "" && translated_map_val != undefined && translated_map_val != "") {
```

Comment:
Expected '===' and instead saw '=='. (eqeqeq)
Expected '!==' and instead saw '!='. (eqeqeq)